documentation/library_guide/parallel_api/pass_data_algorithms.rst

```diff
@@ -1,20 +1,36 @@
 Pass Data to Algorithms
 #######################
 
-When using the C++ standard execution policies, oneDPL supports data being passed to its algorithms as specified
-in the ISO/IEC 14882:2017 standard (commonly called C++17). According to the standard, the calling code
-must prevent data races when using algorithms with parallel execution policies.
+When using the C++ standard aligned (or *host*) execution policies, oneDPL supports data being passed to its
+algorithms as specified in the ISO/IEC 14882:2017 standard (commonly called C++17). According to the standard,
+the calling code must prevent data races when using algorithms with parallel execution policies.
 
-Note: Implementations of ``std::vector<bool>`` are not required to avoid data races for concurrent modifications
-of vector elements. Some implementations may optimize multiple ``bool`` elements into a bitfield, making it unsafe
-for multithreading. For this reason, it is recommended to avoid ``std::vector<bool>`` for anything but a read-only
-input with the C++ standard execution policies.
+.. note::
+   Implementations of ``std::vector<bool>`` are not required to avoid data races for concurrent modifications
+   of vector elements. Some implementations may optimize multiple ``bool`` elements into a bitfield, making it unsafe
+   for multithreading. For this reason, it is recommended to avoid ``std::vector<bool>`` for anything but a read-only
+   input with the C++ standard aligned execution policies.
 
 When using a device execution policy, you can use one of the following ways to pass data to an algorithm:
 
-* ``oneapi::dpl::begin`` and ``oneapi::dpl::end`` functions
-* Unified shared memory (USM) pointers
-* ``std::vector`` with or without a USM allocator
+* ``oneapi::dpl::begin`` and ``oneapi::dpl::end`` functions for a `SYCL buffer
+  <https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#subsec:buffers>`_;
+* `Unified shared memory (USM) <https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:usm>`_ pointers;
+* ``std::vector`` with or without a USM allocator.
+
+For an algorithm to access data, it is important that the used execution policy matches the data storage type.
+The following table shows which execution policies can be used with various data storage types.
+
+====================================== =============== =============
+Data Storage                           Device Policies Host Policies
+====================================== =============== =============
+SYCL buffer                            Yes             No
+Device-allocated USM                   Yes             No
+Shared and host-allocated USM          Yes             Yes
```

Comment:
I am not sure if it makes sense but how about something like 
```suggestion
Shared and host-allocated USM (including USM allocator)          Yes             Yes
```
The reason why I am asking that is because we don't mention `usm_allocator` separately from `std::vector`. To me having `usm_allocator` without `std::vector` is a valid use-case

Reply:
It is a valid use case, however since we mention USM in general and not a specific way of allocation, I think it implicitly covers the allocator as well. The next rows actually focus on the vector, not on the allocator.